ddl/ddl_api.go

```diff
@@ -1550,6 +1550,7 @@ func handleTableOptions(options []*ast.TableOption, tbInfo *model.TableInfo) err
 			if tbInfo.ShardRowIDBits > shardRowIDBitsMax {
 				tbInfo.ShardRowIDBits = shardRowIDBitsMax
 			}
+			tbInfo.MaxShardRowIDBits = tbInfo.ShardRowIDBits
```

Comment:
Can you sure that handleTableOptions is only called by CreateTable in the future? If not, what about:

```
if tbInfo.MaxShardRowIDBits < tbInfo.ShardRowIDBits {
	tbInfo.MaxShardRowIDBits = tbInfo.ShardRowIDBits
}
```